### DISCUSSION TITLE
Update Maven formula to depend on Temurin

### DISCUSSION THREAD

**Discussion by chriskilding** [Homebrew/discussions#2065] [Casks] [answered]
The Maven formula currently depends on the OpenJDK formula from https://github.com/AdoptOpenJDK/homebrew-openjdk:

https://github.com/Homebrew/homebrew-core/blob/ed8100f38b42976612e12a6f8c8501a107e46818/Formula/maven.rb#L22

Recently the OpenJDK formula was deprecated; development was stopped on August 1st. They recommend changing to the Temurin brew casks instead.

Either the Maven formula needs to depend on `temurin` directly, or the `openjdk` formula alias needs to be changed to point at Temurin.

**Comment by SMillerDev** [comment, marked as answer]
It doesn't depend on the AdoptOpenJDK cask. Homebrew never depends on external casks!
It depends on the OpenJDK formula: https://formulae.brew.sh/formula/openjdk

**Reply by chriskilding** [reply]
Right, yes. Think I got confused with all the recent AdoptOpenJDK/Adoptium/Temerin changes...

I first did `brew install --cask temurin`, then `brew install maven`. Rather than noticing I already had a JDK on the system, the Maven install then dragged in the OpenJDK formula as well.

Would it be possible for the Temurin casks to act as `openjdk` implementations as far as brew is concerned? So that if they're already installed, it doesn't try to install another OpenJDK?

**Reply by carlocab** [reply]
Unfortunately not, as that's not a configuration we test in CI. Homebrew used to allow casks to substitute formulae, but that led to a lot of users complaining about things not working.

**Reply by SMillerDev** [reply]
You can however set `JAVA_HOME` to the Temurin cask and then most programs will just use the Temurin install as their Java of choice

**Reply by jkleckner** [reply]
So if openjdk and adoptopenjdk8 are both deprecated, what is the right way to get a jdk8 current install?

Using `brew install temurin@8` doesn't work.
